### PR TITLE
MIGENG-586: Application platform information and recommendations

### PR DESCRIPTION
**Pull request by carlosthe19916**
https://issues.redhat.com/browse/MIGENG-586

Depends on: https://github.com/project-xavier/xavier-analytics/pull/64

Added a new recommended target called "Red Hat JBoss EAP"

### CODE REVIEW

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java [hunk 1]
```diff
@@ -5081,6 +5220,286 @@ public void testDebianFamily() throws ParseException {
...
+        QueryResults queryResults= (QueryResults) results.get(QUERY_IDENTIFIER);
+        Assert.assertEquals(1, queryResults.size());
+
+        QueryResultsRow queryResultsRow = queryResults.iterator().next();
+        Assert.assertThat(queryResultsRow.get("report"), instanceOf(WorkloadInventoryReportModel.class));
+
+        WorkloadInventoryReportModel workloadInventoryReportModel = (WorkloadInventoryReportModel) queryResultsRow.get("report");
+        Assert.assertEquals("CentOS", workloadInventoryReportModel.getOsFamily());
+        Assert.assertTrue(workloadInventoryReportModel.getWorkloads().stream().anyMatch(target -> target.toLowerCase().contains("Tomcat".toLowerCase())));
+        Assert.assertFalse(workloadInventoryReportModel.getRecommendedTargetsIMS().stream().anyMatch(target -> target.toLowerCase().contains("Red Hat JBoss EAP".toLowerCase())));
```
Comment (m-brophy) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java:5478]: Could we add a test for "IBM Websphere App Server" workload causing "Target_RedHatJBossEAP" in WorkloadInventoryReportTest.java?

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/TargetsTest.java [hunk 2]
```diff
@@ -510,4 +510,291 @@ public void testNoOCPTargetHotplugFlagSupportedOS() {
...
+
+        facts.put("workloadInventoryReportModel",workloadInventoryReportModel);
+
+        Map<String, Object> results = createAndExecuteCommandsAndGetResults(facts);
+
+        Assert.assertEquals(3, results.get(NUMBER_OF_FIRED_RULE_KEY));
+        Utils.verifyRulesFiredNames(this.agendaEventListener,
+                "AgendaFocusForTest",
+                "Target_None",
+                "Target_OpenJDK"
```
Comment (m-brophy) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/TargetsTest.java]: The same comments apply as in #64 re: "Target_None" and other targets co-existing. Also, as the changes for that PR are also here, do we need to merge #64 at all?

Comment (carlosthe19916) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/TargetsTest.java]: @m-brophy this should be solved after merging https://github.com/project-xavier/xavier-analytics/pull/64. This PR is based on https://github.com/project-xavier/xavier-analytics/pull/64

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java (hunk 1 above)
Comment (carlosthe19916) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java:5478]: @m-brophy I've just added a new test here https://github.com/carlosthe19916/xavier-analytics/blob/a9fe97eb254af76255e68555a4e859de7e21e576/src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java#L5483

**Pull request merged**